tracer/src/Datadog.Trace/AppSec/Waf/NativeBindings/DdwafRuleSetInfo.cs

```diff
@@ -1,15 +1,21 @@
-// <copyright file="DdwafRuleSetInfoStruct.cs" company="Datadog">
+// <copyright file="DdwafRuleSetInfo.cs" company="Datadog">
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
 using System;
 using System.Runtime.InteropServices;
+#pragma warning disable SA1401
+#pragma warning disable CS0649
 
 namespace Datadog.Trace.AppSec.Waf.NativeBindings
 {
+    /// <summary>
+    /// This must be a class because it can sometimes be null
+    /// the waf only needs it when updating rules. For other updates, this won't be filled and will be misleading
+    /// </summary>
     [StructLayout(LayoutKind.Sequential)]
```

Comment:
If it's a class remove this attribute which will do nothing for classes.

Reply:
They seemed to say it's taken into account whether it's class or struct [here](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.structlayoutattribute?view=net-7.0#remarks)  🤔